### PR TITLE
Displaying the gift subscription in subscriber's page

### PR DESCRIPTION
**Pull request by millerf**
Fixes https://github.com/Automattic/gold/issues/287
Fixes https://github.com/Automattic/gold/issues/288

## Proposed Changes

* Display `Gift: NAME OF THE PLAN` in the subscription column
* Fixes the subscription details page
* Some small code refactor/type def  
* Fixes read/subscriptions/SUBSCRIPTION_ID
* if all payments are gifts, the "cancel subscription" should be disabled
Screenshot 2024-01-24 at 16 36 14

Screenshot 2024-01-24 at 17 23 37
Screenshot 2024-01-24 at 18 26 55

## Testing Instructions

* Apply D135764-code on your sandbox
* Sandbox public-api.wordpress.com
* Go to any site with Gifts enable 
* In the subscribers' page, gift any user.
* The Subscription column should display "undefined ()"
* Access the same page via the calypso live link and you should see `Gift: NAME_OF_THE_GIFTED_PLAN`
* Click on "View"
Screenshot 2024-01-24 at 17 25 14
*  The subscription details should be fine
* Visit http://calypso.localhost:3000/read/subscriptions/ and click on a gifted subscription
* You should see "Gift" for one of the payment, and if all payments are gifts, the "cancel subscription" should be disabled
 
## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/blocks/reader-site-subscription/details.tsx [hunk 1]
```diff
@@ -284,7 +307,7 @@ const SiteSubscriptionDetails = ( {
 						<Button
 							className="site-subscription-page__unsubscribe-button"
 							onClick={ onClickCancelSubscriptionButton }
-							disabled={ unsubscribing }
+							disabled={ unsubscribing || areAllPaymentsGifts }
```
Comment (millerf) [inline client/blocks/reader-site-subscription/details.tsx:310]: This is to prevent unsubscribing when there is actually no paid subscription

**Comment by chrisfromthelc** [comment]
I patched the WPCOM diff and used the Calypso Live link. I see the gift properly formatted in the single user display, but not the Subscribers list.

Single view
Screenshot 2024-01-24 at 12 22 03 PM

Subscriber list view
Screenshot 2024-01-24 at 12 25 33 PM

No console errors. I tried using the patch and running Calypso locally, but get the same behavior.

**Comment by millerf** [comment]
@chrisfromthelc I think I see the issue: it is missing a data refresh when the subscription is gifted:

https://github.com/Automattic/wp-calypso/assets/790558/69393f87-0a84-467e-a117-32da0f2285f7

**Comment by millerf** [comment]
@chrisfromthelc I added a timestamp so the cache is burst and the new set of data is displayed

https://github.com/Automattic/wp-calypso/assets/790558/df44fb22-6e04-48e7-8fd7-ccfdf8f33c13

**Comment by chrisfromthelc** [comment]
Seems like the failing tests are actually an issue. Even tried rebasing locally and still couldn't build it.

```
webpack-client.ts:43 ./my-sites/subscribers/helpers/index.ts undefined
Module build failed (from ../node_modules/thread-loader/dist/cjs.js):
Thread Loader (Worker 3)
/Users/christophersmith/Documents/Development/WordPress/wp-calypso/client/my-sites/subscribers/helpers/index.ts: Argument name clash. (11:1)

   9 | 	filterOption?: string,
  10 | 	timestamp?: number,
> 11 | 	filterOption?: string,
     | 	^
  12 | 	hasManySubscribers?: boolean
  13 | ) => {
  14 | 	const cacheKey = [ 'subscribers', siteId ];
  ```

**Comment by millerf** [comment]
> Seems like the failing tests are actually an issue. Even tried rebasing locally and still couldn't build it.
> 
> ```
> webpack-client.ts:43 ./my-sites/subscribers/helpers/index.ts undefined
> Module build failed (from ../node_modules/thread-loader/dist/cjs.js):
> Thread Loader (Worker 3)
> /Users/christophersmith/Documents/Development/WordPress/wp-calypso/client/my-sites/subscribers/helpers/index.ts: Argument name clash. (11:1)
> 
>    9 | 	filterOption?: string,
>   10 | 	timestamp?: number,
> > 11 | 	filterOption?: string,
>      | 	^
>   12 | 	hasManySubscribers?: boolean
>   13 | ) => {
>   14 | 	const cacheKey = [ 'subscribers', siteId ];
> ```

Funny I tried to build locally yesterdy and it worked!! Anyway,  I'll fix it now.

File: client/my-sites/subscribers/components/subscriber-details/subscriber-details.tsx [hunk 2]
```diff
@@ -88,28 +116,16 @@ const SubscriberDetails = ( {
 						{ subscriptionPlans &&
 							subscriptionPlans.map( ( subscriptionPlan, index ) => (
 								<div className="subscriber-details__content-value" key={ index }>
-									{ subscriptionPlan.title ? `${ subscriptionPlan.title } - ` : '' }
+									{ ! subscriptionPlan.is_gift && subscriptionPlan.title
```
Comment (chrismccluskey) [inline client/my-sites/subscribers/components/subscriber-details/subscriber-details.tsx:119]: Should this also be included in the `SubscriberRow` component? I was expecting to see the gifts in the subscribers list (not just the profile) as you showed in your screenshot but was unable to find this in the calypso live links.

Comment (millerf) [reply client/my-sites/subscribers/components/subscriber-details/subscriber-details.tsx:119]: @chrismccluskey You definitely should see it in the subscriber's list.
Are you sure:
- you are sandboxing
- you have applied D135764-code  on your sandbox?
- you are using `define( 'USE_STORE_SANDBOX', true );` if necessary?

Comment (millerf) [reply client/my-sites/subscribers/components/subscriber-details/subscriber-details.tsx:119]: Screenshot 2024-02-01 at 16 27 10

Comment (millerf) [reply client/my-sites/subscribers/components/subscriber-details/subscriber-details.tsx:119]: Screenshot 2024-02-01 at 17 13 21

Comment (chrismccluskey) [reply client/my-sites/subscribers/components/subscriber-details/subscriber-details.tsx:119]: > you are sandboxing
> you have applied D135764-code on your sandbox?
> you are using define( 'USE_STORE_SANDBOX', true ); if necessary?

Yes to all three of those. I am only not seeing it in the subscribers list:

Screenshot 2024-02-01 at 11 21 35 AM

I can see it when viewing the profile:

Screenshot 2024-02-01 at 11 21 48 AM

This was done with the `container-condescending-almeida.calypso.live` container.

Comment (millerf) [reply client/my-sites/subscribers/components/subscriber-details/subscriber-details.tsx:119]: I see, you have both a gift and a normal subscription...

Comment (millerf) [reply client/my-sites/subscribers/components/subscriber-details/subscriber-details.tsx:119]: Actually I tested it and it works fine, this one has a Gift AND a subscription:

Screenshot 2024-02-02 at 15 19 42

Comment (millerf) [reply client/my-sites/subscribers/components/subscriber-details/subscriber-details.tsx:119]: > Yes to all three of those. I am only not seeing it in the subscribers list:

I am not sure... Does you site has the `MEMBERSHIPS_GIFT_COUPONS_STICKER` sticker?

File: client/my-sites/subscribers/hooks/use-subscription-plans.ts [hunk 3]
```diff
@@ -7,52 +7,66 @@ import {
...
 
 	function formatRenewalPrice( renewalPrice: number, currency: string ) {
 		if ( ! renewalPrice ) {
 			return '';
 		}
-
-		const money = getCurrencyObject( renewalPrice, currency );
-		return money.integer !== '0' ? `${ money.symbol }${ money.integer }` : '';
+		const money = getCurrencyObject( renewalPrice, currency, { stripZeros: false } );
+		return money.hasNonZeroFraction
```
Comment (millerf) [inline client/my-sites/subscribers/hooks/use-subscription-plans.ts:46]: That seems to fix @JuanLucha code and properly display plan prices with decimals (when needed), and prevent displaying an empty string if the price is < 0

**Review by chrismccluskey — APPROVED** [review]
LGTM and works as expected.

**Pull request merged**